### PR TITLE
Split out an EditorFeatures.Wpf assembly

### PR DESCRIPTION
**Pull request by jasonmalinowski**
This splits out an EditorFeatures.Wpf assembly, which contains the parts that depend on IWpfTextView and WPF itself. The parts that depend on the editor APIs but not these are left in EditorFeatures. A follow-up PR will actually remove the dependency on Microsoft.VisualStudio.Text.UI.Wpf, but that must be done with an update to the packages and thus has to happen in the dev15.5.x branch.

I _highly_ recommend you review a commit at a time.

### CODE REVIEW

File: src/EditorFeatures/Core/Shared/Utilities/ForegroundThreadAffinitizedObject.cs [hunk 1]
```diff
@@ -86,17 +79,13 @@ public ForegroundThreadAffinitizedObject(bool assertIsForeground = false)
...
-                CurrentForegroundThreadData.Kind == ForegroundThreadDataKind.Unknown ||
-                Application.Current == null ||
-                Application.Current.Dispatcher.Thread == ForegroundThread);
-
             // ForegroundThreadAffinitizedObject might not necessarily be created on a foreground thread.
             // AssertIsForeground here only if the object must be created on a foreground thread.
             if (assertIsForeground)
             {
+                // Assert we have some kind of foreground thread
+                Contract.ThrowIfFalse(CurrentForegroundThreadData.Kind == ForegroundThreadDataKind.Unknown);
```
Comment (KirillOsenkov) [inline src/EditorFeatures/Core/Shared/Utilities/ForegroundThreadAffinitizedObject.cs]: >Contract.ThrowIfFalse(CurrentForegroundThreadData.Kind == ForegroundThreadDataKind.Unknown); [](start = 16, length = 92)

This is suspicious. Did you mean the opposite?

**Comment by KirillOsenkov** [comment]
👍

**Comment by jasonmalinowski** [comment]
@dotnet-bot retest windows_determinism_prtest please.

**Comment by jasonmalinowski** [comment]
@dotnet-bot retest this please.

**Comment by jasonmalinowski** [comment]
@dotnet-bot retest ubuntu_14_debug_prtest please.

**Pull request merged**